### PR TITLE
support Express Custom Setup

### PR DESCRIPTION
**Pull request by liubing-microsoft**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/IntegrationRuntime.json
Comment (ArcturusZhang) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/IntegrationRuntime.json]: Cmdkey is failing the spell check. Consider add this word to `custom_words.txt`

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]
Semantic and LintDiff checks fail, please fix this.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/IntegrationRuntime.json
Comment (ArcturusZhang) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/IntegrationRuntime.json]: This should be `object`

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/IntegrationRuntime.json
Comment (ArcturusZhang) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/IntegrationRuntime.json]: You must add a `required` field for `discriminator`

Comment (liubing-microsoft) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/IntegrationRuntime.json]: added

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/IntegrationRuntime.json
Comment (liubing-microsoft) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/IntegrationRuntime.json]: added

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/IntegrationRuntime.json
Comment (liubing-microsoft) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/IntegrationRuntime.json]: Cmdkey is added in custom-words.txt

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/IntegrationRuntime.json
Comment (ArcturusZhang) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/IntegrationRuntime.json]: I think this `required` should be `type` as you did in line 260-262

Comment (liubing-microsoft) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/IntegrationRuntime.json]: I have a question about this comment: The reason that `CustomSetupBase` `required` `type` (260-262) is the name of its "properties" has "type". But `ComponentSetupTypeProperties` here has no `type` properties.

Comment (ArcturusZhang) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/IntegrationRuntime.json]: In my knowledge, this `required` field is filled with discriminator, in this case `type` which is the value of `discriminator`. Referring to this [article](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md). I am no expert on this. You could view the error reported by semantic check, and make changes corresponding to that.

Comment (liubing-microsoft) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/IntegrationRuntime.json]: added. Many thanks.

**Comment by liubing-microsoft** [comment]
Hi @ArcturusZhang , I have updated my changes but the semantic check is still failed which is not introduced by my changes. How to handle it?

>  error : 
code: OAV131
error: DISCRIMINATOR_NOT_REQUIRED
name: discriminator
params:
>  - type
message: discriminator must be a required property.
path:
>  - 'definitions,IntegrationRuntimeStatus'

> error : 
code: OAV131
error: DISCRIMINATOR_NOT_REQUIRED
name: discriminator
params:
>  - type
message: discriminator must be a required property.
path:
> - 'definitions,SsisObjectMetadata'

**Comment by ArcturusZhang** [comment]
> Hi @ArcturusZhang , I have updated my changes but the semantic check is still failed which is not introduced by my changes. How to handle it?
> 
> > error :
> > code: OAV131
> > error: DISCRIMINATOR_NOT_REQUIRED
> > name: discriminator
> > params:
> > 
> > * type
> >   message: discriminator must be a required property.
> >   path:
> > * 'definitions,IntegrationRuntimeStatus'
> 
> > error :
> > code: OAV131
> > error: DISCRIMINATOR_NOT_REQUIRED
> > name: discriminator
> > params:
> > 
> > * type
> >   message: discriminator must be a required property.
> >   path:
> > * 'definitions,SsisObjectMetadata'

This is caused by something changed in the semantic tool. I asked owner of CI @NullMDR  how to resolve this, but do not get reply right now.

**Review by hvermis — APPROVED** [review]

**Comment by liubing-microsoft** [comment]
> resolve

@ArcturusZhang , Could you tell me when the semantic tool issue can be resolved? It will block our feature release. Thank you.

**Comment by raych1** [comment]
@liubing-microsoft , the semantic errors was fixed by [PR](https://github.com/Azure/azure-rest-api-specs/pull/6989), please merge the change and re-run semantic check, it should pass.

**Comment by liubing-microsoft** [comment]
> Semantic and LintDiff checks fail, please fix this.

Hi @ArcturusZhang , all checks have passed.

**Review by ArcturusZhang — APPROVED** [review]

**Comment by ArcturusZhang** [comment]
> > Semantic and LintDiff checks fail, please fix this.
> 
> Hi @ArcturusZhang , all checks have passed.

I approved this. But please wait for ARM team response before you merge this

**Comment by ravbhatnagar** [comment]
this is typically a breaking change, but for ADF the scenario is a bit more nuanced. We have discussed this in person in the past and given them exception with the understanding that we will bring up this usecase in front of folks like MarkRuss and seek guidance. Signing off from ARM side.

**Pull request merged**
